modules/spellfiles.nix

```diff
@@ -0,0 +1,175 @@
+
+        See https://neovim.io/doc/user/spell.html#spell-NOSUGFILE.
+      '';
+    };
+
+    spellfiles = lib.mkOption {
+      type = with types; listOf (enum availableSpellfiles);
+      default = [ ];
+      description = ''
+        Explicitly specify spellfiles to install.
+
+        Possible values correspond to the spellfiles available at https://ftp.nluug.nl/pub/vim/runtime/spell/.
+      '';
+      example = [
+        "en.utf-8.spl"
+        "en.utf-8.sug"
+        "fr.utf-8.spl"
+      ];
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    extraFiles =
+      let
+        /*
+          Infers the list of language ids from `opts.spelllang`:
+
+          Turns: "en_us,fr,de"
+          into: [ "en" "fr" "de" ]
+          # https://neovim.io/doc/user/spell.html#spell-load
+        */
+        languagesFromSpelllang = lib.pipe (config.spelllang or "en") [
+
+          # "en_us,fr,de" -> [ "en_us" "fr" "de" ]
+          (lib.splitString ",")
+
+          # Remove region from language id ("en_us" -> "en")
+          (map (lang: lib.head (builtins.match "^([^_]+)" lang)))
+        ];
+        languages = if cfg.languages == "auto" then languagesFromSpelllang else cfg.languages;
+
+        /*
+          Infers the spellfiles names from a language identifier.
+
+          The spellfiles associated with language `LL` are:
+          - `LL.<ENCODING>.spl`: main spell file
+          - `LL.<ENCODING>.sug` (optional): additional file for more precise suggestions
+
+          `<ENCODING>` is inferred from `opts.encoding` (if set) and defaults to `utf-8`.
+        */
+        getFilenamesFromLanguage =
+          lang:
+          let
+            # Default to utf-8 encoding
+            encoding = config.opts.encoding or "utf-8";
+
+            # Main spellfile (.spl)
+            baseName = "${lang}.${encoding}";
+
+            # Optional suggestion spellfile (.sug)
+            sugFilename = "${baseName}.sug";
+            includeSugSpellfile =
+              # user has enabled fetching suggestion spellfiles
+              cfg.includeSuggestions
+
+              # a suggestion spellfile is available for this language-encoding pair
+              && (lib.hasAttr sugFilename spellfilesHashes);
+          in
+          [
+            "${baseName}.spl"
+          ]
+          ++ lib.optional includeSugSpellfile sugFilename;
+
+        # [ "en.utf-8.spl" "en.utf-8.sug" "fr.utf-8.spl" ]
+        spellfiles = lib.unique (
+          # spellfiles from `cfg.languages`
+          (lib.concatMap getFilenamesFromLanguage languages)
+
+          # explicitly requested spellfiles
+          ++ cfg.spellfiles
+        );
+
+        /*
+          Maps "en.utf-8.spl" to:
+          {
+            name = "spell/en.utf-8.spl";
+
+            value.source = pkgs.fetchurl {
+              url = "https://ftp.nluug.nl/pub/vim/runtime/spell/en.utf-8.spl";
+              hash = "sha256-/sq9yUm2o50ywImfolReqyXmPy7QozxK0VEUJjhNMHA=";
+            };
+          }
```

Comment:
I'm a little concerned about our pinning technique.

Firstly, I assume that spell files are occasionally modified, and don't offer a versioned/pinned mirror?

This leads to my issue with the assumption that we can generate a table of file hashes without running into issues.

E.g. a user rolls back to an old nixvim revision: now the hash is outdated \
Or: nixvim is slow to update: same outcome... \
Or: someone is bisecting their config \
Or: we are bisecting nixvim (which has a test for spell files)

I think we either need to download the actual spell files in our generate script, or find a way to have versioned URLs.